sycl/test/basic_tests/memory-consumption.cpp

```diff
@@ -0,0 +1,69 @@
+// RUN: %clangxx -fsycl %s -o %t.out
+// RUN: env SYCL_BE=%sycl_be %t.out
+// RUN: %t.out
+//
+//
+//==-----memory-consumption.cpp - SYCL memory consumption basic test ------==//
+//
+// This test specifically tests that memory consumption does not change
+// when the get_devices() is called repeatedly.
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "CL/sycl.hpp"
+#include <iostream>
+#include <thread>
+
+#ifdef __linux__
+#include <strings.h>
+#include <sys/resource.h>
+#include <sys/time.h>
+
+long get_cpu_mem() {
+  struct rusage usage;
+  memset(&usage, 0, sizeof(rusage));
+  getrusage(RUSAGE_SELF, &usage);
+  return usage.ru_maxrss;
+}
+#endif
+
+using namespace cl::sycl;
+
+int main() {
+  constexpr auto dev_type = info::device_type::gpu;
+  auto devices = device::get_devices(dev_type);
+
+#ifdef __linux__
+  int startSize = get_cpu_mem();
+  std::cout << startSize << " kb" << std::endl;
+
+  for (int i = 0; i < 1000; i++) {
+    devices = cl::sycl::device::get_devices(dev_type);
+  }
+  int endSize = get_cpu_mem();
+  std::cout << endSize << " kb" << std::endl;
+#else
+  for (int i = 0; i < 1000; i++) {
```

Comment:
We do not check memory for windows is there any reason to run the test on windows?
I would prefer to disable it there by putting something like
// No way to count memory
// UNSUPPORTED: windows

Reply:
Done

Reply:
please remove dead code as well